server/sandbox_stop_linux.go

```diff
@@ -112,8 +112,11 @@ func (s *Server) stopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 	if err := s.StorageRuntimeServer().StopContainer(sb.ID()); err != nil && !errors.Is(err, storage.ErrContainerUnknown) {
 		log.Warnf(ctx, "failed to stop sandbox container in pod sandbox %s: %v", sb.ID(), err)
 	}
-	if err := s.ContainerStateToDisk(podInfraContainer); err != nil {
-		log.Warnf(ctx, "error writing pod infra container %q state to disk: %v", podInfraContainer.ID(), err)
+
+	if podInfraContainer != nil {
+		if err := s.ContainerStateToDisk(podInfraContainer); err != nil {
+			log.Warnf(ctx, "error writing pod infra container %q state to disk: %v", podInfraContainer.ID(), err)
+		}
```

Comment:
Looks like a naked check at line 70 too:  `if ctr.ID() == podInfraContainer.ID() {`

Reply:
And also quite similar to: https://github.com/cri-o/cri-o/blob/49f667a77e393ab80e0725a9f3bf00fd89e7fcc9/server/sandbox_stop_linux.go#L99

Reply:
It looks like we're not expecting `podInfraContainer == nil` in here, so maybe just add a `!= nil` check after line 56 -- and error out if it's `nil`, removing all the subsequent checks?

Reply:
I am thinking we should just handle this in ContainerStateToDisk() instead of here